stacks-signer/src/signerdb.rs

```diff
@@ -16,15 +16,64 @@
 
 use std::path::Path;
 
+use blockstack_lib::chainstate::nakamoto::{NakamotoBlock, NakamotoBlockVote};
 use blockstack_lib::util_lib::db::{
     query_row, sqlite_open, table_exists, u64_to_sql, Error as DBError,
 };
+use libsigner::BlockProposal;
 use rusqlite::{params, Connection, Error as SqliteError, OpenFlags, NO_PARAMS};
+use serde::{Deserialize, Serialize};
 use slog::slog_debug;
 use stacks_common::debug;
 use stacks_common::util::hash::Sha512Trunc256Sum;
+use wsts::net::NonceRequest;
+
+/// Additional Info about a proposed block
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct BlockInfo {
```

Comment:
I think this is mostly v1 data?  If so, then perhaps `BlockInfo` ought to be an `enum` of `v0` and `v1` variants?

Reply:
Yes there is not much difference between the two. Just NonceRequest (optional) and the bool "signed_over" which is used to prevent multiple signing rounds. I am not against making an entirely separate table, but it seemed a bit overkill.

Reply:
@hstove You can still do that; you'd just have explicit getters and setters for the variants' inner records, instead of directly accessing them.  But adding this would perhaps be needlessly tedious.

Instead, perhaps the v0- and v1-specific state can be captured in an enum, and leave the rest of the struct (which is common to both versions) untouched?

Reply:
Captured in a ticket for now https://github.com/stacks-network/stacks-core/issues/4816